openlibrary/core/models.py

```diff
@@ -428,18 +432,25 @@ def from_isbn(cls, isbn: str) -> "Edition | None":  # type: ignore[return]
         # Finally, try to fetch the book data from Amazon + import.
         try:
             if metadata := get_amazon_metadata(
-                id_=isbn10 or isbn13, id_type="isbn", high_priority=True
+                id_=isbn10 or isbn13, id_type="isbn", high_priority=high_priority
             ):
-                cleaned_metadata = clean_amazon_metadata_for_load(metadata)
-                import_item = {
-                    'ia_id': cleaned_metadata['source_records'][0],
-                    'status': 'staged',
-                    'data': cleaned_metadata,
-                }
-                batch = get_current_amazon_batch()
-                batch.add_items([import_item])
                 if edition := ImportItem.import_first_staged(identifiers=isbns):
                     return edition
+                else:
+                    # This is almost certainly duplicative; the affiliate-server
+                    # will queue for import as `staged` any Product API responses
+                    # that are put into the cache before returning metedata to
+                    # get_amazon_metadata(). But perhaps the DB is slow.
+                    cleaned_metadata = clean_amazon_metadata_for_load(metadata)
+                    import_item = {
+                        'ia_id': cleaned_metadata['source_records'][0],
+                        'status': 'staged',
+                        'data': cleaned_metadata,
+                    }
+                    batch = get_current_amazon_batch()
```

Comment:
Instead of get_current_amazon_batch, let's assume there will be a single batch for amazon, keyed by amazon id / isbn.